### PR TITLE
[Widget]Localizaton for unreadmessages count

### PR DESCRIPTION
**Pull request by dishakhattri**

### CODE REVIEW

File: chat-components/src/components/chatbutton/ChatButton.tsx [hunk 1]
```diff
@@ -19,15 +19,26 @@ function NotificationBubble(props: IChatButtonProps, parentId: string) {
         root: Object.assign({}, defaultChatButtonNotificationBubbleStyles, props.styleProps?.notificationBubbleStyleProps)
     };
 
+    const ihiddenTextStyles: React.CSSProperties = {
```
Comment (charliewang95) [inline chat-components/src/components/chatbutton/ChatButton.tsx:22]: We are reverting the component changes after the other PR checks in right?

File: chat-widget/src/components/webchatcontainerstateful/WebChatContainerStateful.tsx [hunk 2]
```diff
@@ -26,13 +27,15 @@ export const WebChatContainerStateful = (props: IWebChatContainerStatefulProps)
     
     const localizedTexts = {
         ...defaultMiddlewareLocalizedTexts,
+        ...defaultChatButtonLocalizedTexts,
```
Comment (charliewang95) [inline chat-widget/src/components/webchatcontainerstateful/WebChatContainerStateful.tsx:30]: Please don't add complexities in WebChatContainer. The localized texts for Chat Button should come as controlProps for the ChatButton component itself

File: chat-widget/src/components/chatbuttonstateful/common/defaultProps/defaultChatButtonLocalizedTexts.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { ILiveChatWidgetLocalizedTexts } from "../../../../contexts/common/ILiveChatWidgetLocalizedTexts";
+
+export const defaultChatButtonLocalizedTexts: ILiveChatWidgetLocalizedTexts = {
```
Comment (charliewang95) [inline chat-widget/src/components/chatbuttonstateful/common/defaultProps/defaultChatButtonLocalizedTexts.ts:3]: Remove this file per the other comment

File: chat-widget/src/contexts/common/ILiveChatWidgetContext.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@ export interface ILiveChatWidgetContext {
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
         postChatContext: any;
         middlewareLocalizedTexts: ILiveChatWidgetLocalizedTexts | undefined;
+        chatbuttonLocalizedTexts: ILiveChatWidgetLocalizedTexts | undefined;
```
Comment (charliewang95) [inline chat-widget/src/contexts/common/ILiveChatWidgetContext.ts:18]: chat[B]utton

File: chat-widget/src/contexts/common/LiveChatWidgetActionType.ts [hunk 5]
```diff
@@ -18,6 +18,12 @@ export enum LiveChatWidgetActionType {
     */
     SET_MIDDLEWARE_LOCALIZED_TEXTS,
 
+    /*
+        Parameters:
+        - props: The localized middlware texts for bhat button
+    */
+        SET_CHATBUTTON_LOCALIZED_TEXTS,
```
Comment (charliewang95) [inline chat-widget/src/contexts/common/LiveChatWidgetActionType.ts:25]: nit: indent

File: chat-components/src/components/chatbutton/ChatButton.tsx (hunk 1 above)
Comment (dishakhattri) [reply chat-components/src/components/chatbutton/ChatButton.tsx:22]: yes, this won't show once the components PR is in.

**Review by charliewang95 — APPROVED** [review]

**Pull request merged**